internal/states/encryption/encryptionconfig/config.go

```diff
@@ -0,0 +1,242 @@
+//
+// When using shared state and shared encryption in this fashion, the author of the other project must pay attention
+// to only expose information that is intended for the current environment in the state.
+type Key string
+
+const (
+	// KeyDefaultRemote contains the configuration key for default values applied only to remote state encryption,
+	// and only if the user provided no other configuration.
+	KeyDefaultRemote Key = "default_remote"
+	// KeyBackend contains the configuration key for overriding the default value when uploading the state to a
+	// remote backend.
+	KeyBackend Key = "backend"
+	// KeyStateFile contains the configuration key for encryption configuration for local state files. The default
+	// remote configuration is not applied to this encryption.
+	KeyStateFile Key = "statefile"
+	// KeyPlanFile contains the configuration key for encryption configuration for local plan files. The default
+	// remote configuration is not applied to this encryption.
+	KeyPlanFile Key = "planfile"
+)
+
+// UsesRemoteDefaults returns true if KeyDefaultRemote should be applied in addition to the current key.
+func (k Key) UsesRemoteDefaults() bool {
+	return k != KeyDefaultRemote && k != KeyStateFile && k != KeyPlanFile
+}
+
+// Validate checks if the key is a valid configuration key.
+func (k Key) Validate() error {
+	switch k {
+	case KeyDefaultRemote:
+		return nil
+	case KeyBackend:
+		return nil
+	case KeyPlanFile:
+		return nil
+	case KeyStateFile:
+		return nil
+	default:
+		if !k.IsRemoteDataSource() {
+			return fmt.Errorf(
+				"invalid encryption configuration key: %s (must be one of %s or contain a dot to specify a remote state data source)",
+				k,
+				strings.Join(
+					[]string{
+						string(KeyDefaultRemote), string(KeyBackend), string(KeyPlanFile), string(KeyStateFile),
+					}, ", ",
+				),
+			)
+		}
+		return nil
+	}
+}
+
+var remoteDatasourceRe = regexp.MustCompile(`^[a-z0-9_]+\.[a-z0-9_]+(|\[(\d+|[a-z0-9_]+)])$`)
+
+// IsRemoteDataSource returns true if the specified key is a valid remote data source key.
+func (k Key) IsRemoteDataSource() bool {
+	return remoteDatasourceRe.MatchString(string(k))
+}
+
+// ConfigEnvName is the name of the environment variable used to configure encryption and decryption as an alternative
+// to providing the configuration in the .tf files directly.
+//
+// Set this environment variable to a JSON representation of ConfigEnvStructure, or leave it unset/blank
+// to disable encryption (default behaviour). If you do not specify a configuration but "enforced" is set to true, tofu
+// will refuse to function. If you specify an invalid JSON, the entire tofu run will fail regardless of the "enforced"
+// setting.
+//
+// Note: With rare exceptions, you should avoid setting the state encryption environment variables in tests,
+// as this may make tests depend on each other. See the comments on encryption.ParseEnvironmentVariables().
+const ConfigEnvName = "TF_STATE_ENCRYPTION"
```

Comment:
The comment here and the comment over `FallbackConfigEnvName` to refer to some things that I can't seem to find

I couldn't find `encryption.ParseEnvironmentVariables` nor `ConfigEnvStructure`

Reply:
Thank you, this is a leftover from the refactoring, fixing it.